### DISCUSSION TITLE
checking for error types

### DISCUSSION THREAD

**Discussion by ahmafi** [panva/jose#519] [Q&A] [answered]
How can I do something like this:

```js
  try {
    const { payload } = await jwtVerify(token, EMAIL_VERIFICATION_JWT_UINT);
  } catch (error) {
    if (error instanceof JWSSignatureVerificationFailed) {
      console.warn(
        'email verification JWT verification failed. Token: %s',
        token
      );
    } else {
      ...
    }
  }
```

For now, I can't import `JWSSignatureVerificationFailed` because it will throw this error:
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/types/util/errors' is not defined by "exports"
```

I'm using `tsx` and importing it like `import { JWSSignatureVerificationFailed } from 'jose/dist/types/util/errors'`.

I can also check for the error code, but I don't think that's a good, because the code is a string and it might change over time.

**Comment by panva** [comment, marked as answer]
Why don't you import just like the project documentation says jose import should be done?

**Reply by ahmafi** [reply]
I couldn't find what you are saying in the docs. Can you please give me a link?

**Reply by panva** [reply]
A link to the readme of this project? https://github.com/panva/jose#readme better yet to how you import the library https://github.com/panva/jose#documentation

BTW feel free to use the codes, they are as stable as the error classes themselves, their presence to avoid having to compare classes is their entire point.

**Reply by ahmafi** [reply]
Importing like `import * as jose from 'jose'` also doesn't work:
![image](https://user-images.githubusercontent.com/12560109/227378342-e4d3a5db-d7de-46e5-995e-4d991fb8403f.png)

The codes are also ok, but I won't get auto-completion with those.

**Reply by panva** [reply]
```ts
import * as jose from 'jose'

// ...

if (x instanceof jose.errors.JWSSignatureVerificationFailed) {

}

```
